internal/config/properties.go

```diff
@@ -80,48 +84,72 @@ func LoadProperties(env string, cm ...*v1.ConfigMap) error {
 		return fmt.Errorf("config map cannot be nil")
 	}
 
-	var awsAccountID string
-	var err error
+	allowedPolicyAction := strings.Split(cm[0].Data[propertyIamPolicyWhitelist], separator)
+	restrictedPolicyResources := strings.Split(cm[0].Data[propertyIamPolicyBlacklist], separator)
+	restrictedS3Resources := strings.Split(cm[0].Data[propertyIamPolicyS3Restricted], separator)
+	awsMasterRole := cm[0].Data[propertyAwsMasterRole]
+
+	Props = &Properties{
+		allowedPolicyAction:       allowedPolicyAction,
+		restrictedPolicyResources: restrictedPolicyResources,
+		restrictedS3Resources:     restrictedS3Resources,
+		awsMasterRole:             awsMasterRole,
+	}
+
+	//Defaults
+	isWebhook := cm[0].Data[propertWebhookEnabled]
+	if isWebhook != "" {
+		Props.isWebhookEnabled = isWebhook
+	} else {
+		Props.isWebhookEnabled = "false"
+	}
 
 	awsRegion := cm[0].Data[propertyAwsRegion]
+	if awsRegion != "" {
+		Props.awsRegion = awsRegion
+	} else {
+		Props.awsRegion = "us-west-2"
+	}
 
-	// Load AWS account ID
-	if Props != nil && Props.awsAccountID != "" {
-		awsAccountID = Props.awsAccountID
+	maxRolesAllowed := cm[0].Data[propertyMaxIamRoles]
+	if maxRolesAllowed != "" {
+		maxRolesAllowed, err := strconv.Atoi(maxRolesAllowed)
+		if err != nil {
+			return err
+		}
+		Props.maxRolesAllowed = maxRolesAllowed
 	} else {
-		awsAccountID, err = awsapi.NewSTS(awsRegion).GetAccountID(context.Background())
+		Props.maxRolesAllowed = 1
+	}
+
+	awsAccountID := cm[0].Data[propertAWSAccountID]
+	// Load AWS account ID
+	if Props.awsAccountID == "" && awsAccountID == "" {
+		awsAccountID, err := awsapi.NewSTS(Props.awsRegion).GetAccountID(context.Background())
 		if err != nil {
 			return err
 		}
+		Props.awsAccountID = awsAccountID
+	} else {
+		Props.awsAccountID = awsAccountID
 	}
 
-	allowedPolicyAction := strings.Split(cm[0].Data[propertyIamPolicyWhitelist], separator)
-	restrictedPolicyResources := strings.Split(cm[0].Data[propertyIamPolicyBlacklist], separator)
-	restrictedS3Resources := strings.Split(cm[0].Data[propertyIamPolicyS3Restricted], separator)
-	awsMasterRole := cm[0].Data[propertyAwsMasterRole]
+	managedPermissionBoundaryPolicy := cm[0].Data[propertyPermissionBoundary]
+
+	if managedPermissionBoundaryPolicy == "" {
+		managedPermissionBoundaryPolicy = "k8s-iam-manager-cluster-permission-boundary"
```

Comment:
shouldn't we make default permission boundary name to be unique per cluster like `k8s-iam-manager-{cluster-name}-permission-boundary`

Reply:
No. By default, we don't need to restrict users to have cluster name in the permission boundary. There will be an issue only if 2 clusters in the same account. So, we consider 2 clusters usecase as corner use case but default usecase is single cluster per account. I might need to update the default param value for CFN if not done